### DISCUSSION TITLE
Clarification on Withdrawal Logic in Foundry FundMe Lesson

### DISCUSSION THREAD

**Discussion by aknokhwal** [Cyfrin/foundry-full-course-cu#1052] [Q&A] [answered]
In the Foundry FundMe lesson provided by Patrick, there's a specific test case testWithDrawFromMultipleFunders where a for loop is used to have 9 users send funds to the FundMe contract. However, I have a question regarding the subsequent usage of the fundMe.Withdraw() function in this scenario.

After all 9 users send funds, the test case only invokes the fundMe.Withdraw() function once. I'm curious about the rationale behind this choice. Shouldn't fundMe.Withdraw() be called 9 times, once for each user who contributed?

I'd appreciate some clarification on why only a single withdrawal is performed in this specific test case, as it might help me better understand the contract's behavior.

Thank you!

```
    function testWithDrawFromMultipleFunders() public funded {
        uint160 numberOfFunder = 10;
        uint160 startingFunderIndex = 1;
        for (uint160 i = startingFunderIndex; i < numberOfFunder; i++) {
            hoax(address(i), sendValue);
            fundMe.fund{value: sendValue}();
        }

        uint256 ownerStartingBalance = fundMe.getOwner().balance;
        uint256 fundMeStartingBalance = address(fundMe).balance;

        vm.startPrank(fundMe.getOwner());
        fundMe.withdraw();
        vm.stopPrank();

        assert(address(fundMe).balance == 0);
        assert(
            fundMeStartingBalance + ownerStartingBalance ==
                fundMe.getOwner().balance
        );
    }
```

**Comment by aknokhwal** [comment]
and also please explain the hoax

**Comment by Niferu** [comment, marked as answer]
Hello @aknokhwal 

If you take a look at the `withdraw` function code you will see this line:

```solidity
(bool success, ) = i_owner.call{value: address(this).balance}("");
```

which means all the funds from contract are going to contract owner. That is why in test we do have only one transaction and not 9. If you would like to give back funds to all users you would need to create loop which calls withdraw each time it find user who funded contract, but it is just not the contract logic from course. The point of this contract is some kind of fund rising for creator where people donates to him and he can withdraw those donates.

**Reply by aknokhwal** [reply]
Thank you for your prompt response and clarification! Your explanation makes sense, and I now understand the reasoning behind calling `fundMe.Withdraw()` once in the test case.

Your assistance is greatly appreciated. Thanks again!

**Reply by Niferu** [reply]
I'm glad I could help! Cheers! :)
